soak-tests/payments-activation.js

```diff
@@ -1,27 +1,160 @@
+                "pan": "5111114000023477",
+                "securityCode": "123"
+            },
+            "idPagamentoFromEC": "e1283f0e673b4789a2af87fd9b4043f4",
+            "type": "CREDIT_CARD"
+        }
+    };
+    const addWalletResponse = http.post(`${paymentManagerApiUrl}/wallet`, addWalletBody, paymentManagerheadersParams, {tags: tagAddWallet});
+    check(addWalletResponse, {'addWallet response is 200': r => r.status === 200}, {tags: tagAddWallet});
+
+    if (addWalletResponse.status !== 200) {
+        console.log("addWallet: " + addWalletResponse.status);
+        console.log(JSON.stringify(addWalletResponse.json()));
+
+        return;
+    }
+
+    const { idWallet } = addWalletResponse.json();
+    const tagPay3ds2 = {pagoPaMethod: "pay3ds2"};
+    const threedsData = {
+        "browserJavaEnabled": "false",
+        "browserLanguage": "it-IT",
+        "browserColorDepth": "24",
+        "browserScreenHeight": "1080",
+        "browserScreenWidth": "1920",
+        "browserTZ": "-120",
+        "browserAcceptHeader": "",
+        "browserIP": "",
+        "browserUserAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:98.0) Gecko/20100101 Firefox/98.0",
+        "acctID": "ACCT_94187",
+        "deliveryEmailAddress": "foo@example.com",
+        "mobilePhone": null
+    };
+    const pay3ds2Body = {
+        "data": {
+            "cvv": "123",
+            "idWallet": 94187,
+            "threeDSData": JSON.stringify(threedsData),
+            "tipo": "web"
+        }
+    };
+    const payResponse = http.post(`${paymentManagerApiUrl}/payments/${idPayment}/actions/pay3ds2`, pay3ds2Body, paymentManagerheadersParams, { tags: tagPay3ds2 });
+    check(payResponse, {'pay3ds2 response is 200': r => r.status === 200}, {tags: tagPay3ds2});
+
+    if (payResponse.status !== 200) {
+        console.log("pay3ds2: " + payResponse.status);
+        console.log(JSON.stringify(payResponse.json()));
+    }
+}
 
 export function setup() {
+    let urlBasePath;
+    if (__ENV.API_VERSION === "v1") {
+        console.info("Running soak tests on Checkout API v1");
+        urlBasePath = setupV1();
+    } else {
+        console.info("Running soak tests on Checkout API v2");
+        urlBasePath = setupV2();
+    }
+
     return {
+        activationUrl: `${urlBasePath}/payment-activations`,
+        verifyUrl: `${urlBasePath}/payment-requests`,
+        activationStatusUrl: `${urlBasePath}/payment-activations`
     };
 }
 
-export default function () {
+export default function ({ verifyUrl, activationUrl, activationStatusUrl }) {
+    const auxDigit = 3;
+    const segregationCode = "00";
+    const idDonation = "00";
+    const iuv = new Date().getTime();
+    const ecCf = 13669721006;
 
-    const min = 302001000000000000;
-    const max = 302001999999999999;
-    const urlBasePath = "https://api.uat.platform.pagopa.it";
-    const rptId = `77777777777${Math.floor(Math.random() * (max - min + 1) + min)}`;
```

Comment:
maybe we leave the possibility to test with mockEC (maybe one day it will be useful :-) )